docs/ci_cd_onboarding.md

```diff
@@ -0,0 +1,11 @@
+# Building Matter
+## Checkout
+```
+git clone --recurse-submodules git@github.com:project-chip/connectedhomeip.git
```

Comment:
Doing recurse-submodules is generally a _bad_ idea for Matter.  Note that we already have docs about how to build.  Why are we adding another one here?  This is the part that is changes requested.

Reply:
I was told to take some notes so it would be easy for new CI/CD hires to get up to speed, but it's true there's nothing too substantial right now. Better to just point to the build docs anyway. I've deleted the file for now.